lib/rules/max-lines.js

```diff
@@ -127,14 +140,35 @@ module.exports = {
                 if (skipComments) {
                     const comments = sourceCode.getAllComments();
 
-                    const commentLines = lodash.flatten(comments.map(comment => getLinesWithoutCode(comment)));
+                    const commentLines = lodash.flatten(
+                        comments.map(comment => getLinesWithoutCode(comment))
+                    );
 
-                    lines = lines.filter(l => !lodash.includes(commentLines, l.lineNumber));
+                    lines = lines.filter(
+                        l => !lodash.includes(commentLines, l.lineNumber)
+                    );
                 }
 
+                lines.forEach((line, i) => {
+                    if (max + 1 === i + 1) {
+                        startLine = line.lineNumber;
+                    }
+                });
+
+                const loc = {
+                    start: {
+                        line: startLine,
+                        column: 0
+                    },
+                    end: {
+                        line: sourceCode.lines.length + 1,
+                        column: 0
+                    }
```

Comment:
This is an invalid location. If we want to report the last location in the source code as the `end` loc, it could be something like:

```js
end: {
    line: sourceCode.lines.length,
    column: lodash.last(sourceCode.lines).length
}
```